tabpy-server/server_tests/test_pwd_file.py

```diff
@@ -0,0 +1,107 @@
+import logging
+import os
+import unittest
+from argparse import Namespace
+from tempfile import NamedTemporaryFile
+
+from tabpy_server.app.app import TabPyApp
+from tabpy_server.app.ConfigParameters import ConfigParameters
+
+try:
+    from unittest.mock import patch, call
+except ImportError:
+    from mock import patch, call
+
+
+class TestPasswordFile(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(TestPasswordFile, self).__init__(*args, **kwargs)
+        self.config_file = None
+        self.pwd_file = None
+        self.cwd = os.getcwd()
+        self.tabpy_cwd = os.path.join(self.cwd, 'tabpy-server', 'tabpy_server')
+
+    def setUp(self):
+        os.chdir(self.tabpy_cwd)
+        self.config_file = NamedTemporaryFile(mode='w', delete=False)
+        self.config_file.close()
+        self.pwd_file = NamedTemporaryFile(mode='w', delete=False)
+        self.pwd_file.close()
+
+    def tearDown(self):
+        os.chdir(self.cwd)
+        os.remove(self.config_file.name)
+        self.config_file = None
+        os.remove(self.pwd_file.name)
+        self.pwd_file = None
+
+    def _set_file(self, file_name, value):
+        with open(file_name, 'w') as f:
+            f.write(value)
+
+    def test_given_no_pwd_file_expect_empty_credentials_list(self):
+        self._set_file(self.config_file.name,
+            "[TabPy]\n"
+            "TABPY_TRANSFER_PROTOCOL = http")
+
+        app = TabPyApp(self.config_file.name)
+        self.assertDictEqual(app.credentials, {}, 'Expected no credentials with no password file provided')
+
+    
+    def test_given_empty_pwd_file_expect_app_fails(self):
+        self._set_file(self.config_file.name,
+            "[TabPy]\n"
+            "TABPY_PWD_FILE = {}".format(self.pwd_file.name))
+        
+        self._set_file(self.pwd_file.name, "# just a comment")
+
+        with self.assertRaises(RuntimeError) as cm:
+            TabPyApp(self.config_file.name)
+            ex = cm.exception
+            self.assertEqual('Failed to read password file {}'.format(self.pwd_file.name), ex.args[0])

```

Comment:
Cleaner to just use `self.assertRaises`

Reply:
This is what happens here:
- L58: `self.assertRaises` checks `RuntimeError` is thrown on `TabPyApp` construction
- L61: check the exception has message we expect to see